web/store/message.ts

```diff
@@ -172,6 +178,37 @@ export const msgStore = createStore<MsgState>('messages', {
         console.log(res.result)
       }
     },
+    async *importMessages({ msgs }, props: NewMsgImport, onSuccess?: Function) {
+      const newChatPayload: NewChat = {
+        name: 'Imported Chat',
+        scenario: props.char.scenario,
+        sampleChat: props.char.sampleChat,
+        greeting: '',
+        overrides: props.char.persona,
+      }
+
+      let chatId: string | undefined
+      try {
+        const newChatPromise = new Promise<string>((resolve, reject) => {
+          const rejectTimeout = setTimeout(reject, 5000)
+          chatStore.createChat(props.char._id, newChatPayload, (chatId) => {
+            clearTimeout(rejectTimeout)
+            resolve(chatId)
+          })
+        })
+        chatId = await newChatPromise
```

Comment:
This feels pretty gross, sorry. Open to suggestions, I needed the chat to be created first to get its ID but store actions aren't async.